src/Indexer.php

```diff
@@ -218,6 +223,53 @@ private function indexStartsWith(string $index, array $indexPrefixes): bool
         return false;
     }
 
+    /**
```

Comment:
It feels to me like the population actions should be handled by their own service, especially when you start doing if()s on the type of handler in the above method